src/psyclone/f2pygen.py

```diff
@@ -911,6 +913,17 @@ def _check_initial_values(dtype, values):
                     raise RuntimeError(
                         "Initial value of '{0}' for a real variable is "
                         "invalid or unsupported".format(val))
+        elif dtype == "character":
+            # Can be a quoted string or a valid Fortran name
```

Comment:
It feels wrong for us to have to quote a string in a string.

Reply:
Yes, I wasn't too happy about it either but I couldn't see any other way of doing it. I could put back the `dtype` argument for `CharDeclGen` and allow it to be e.g. `char_literal` if the supplied string is to be interpreted as a literal string. However, that would then have to apply to the whole list of initial values.